### ISSUE TITLE
Attempting to Load Multiple ROMs Crashes the Program

### ISSUE THREAD

**Issue by Vivit-R** [mgba-emu/mgba#1724]
When I attempt to open any ROM via the File menu while another ROM is already open, the program crashes, printing the following message to stderr:
```
Cannot make QOpenGLContext current in a different thread
Aborted (core dumped)
```

#### Version Information:
0.8.1
Branch: heads/0.8.1
Revision: a16d5f32b2b30944882c557499e8a0e28d784445

I am running the Qt version in X on Ubuntu 18.04.4 LTS.

**Comment by endrift** [comment]
Duplicate of #1642, but I'm unsatisfied with the fix for that (it caused #1686)

**Comment by endrift** [comment]
Fixed in 74edd96 on Windows and GLX. Still need to test on macOS and EGL.

**Comment by endrift** [comment]
Additional fixes in cb7f150cc2b647ae5a9dca2175bd5464d7d5e42f, should work everywhere now, potentially excluding EGL/Wayland.
